Improved Icons for settings ? elements

Rishabh-792 opened the pull request:
PR for "Improve help icons in settings" #20484

**Testing plan:** 
Manual Checks and changed accordingly

**GIFs or screenshots:**
In Light mode :
![Screenshot from 2021-12-08 07-45-48](https://user-images.githubusercontent.com/79650357/145137783-5795f7fe-7640-4d44-b546-f5ee9d8b5c76.png)

In Dark mode :
![Screenshot from 2021-12-08 07-45-55](https://user-images.githubusercontent.com/79650357/145137851-9cb4aefb-5322-4fd0-8540-401960e09c30.png)


New code playground :
![Screenshot from 2021-12-08 07-45-44](https://user-images.githubusercontent.com/79650357/145137910-e46ab08e-93d0-469e-b5b9-b3ba944fedbf.png)
![Screenshot from 2021-12-08 08-01-58](https://user-images.githubusercontent.com/79650357/145138019-0d17c717-7f9f-4609-8039-5689627f7658.png)



<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

alya commented:
The screenshots look reasonable. I'm not sure why this issue is double-assigned... @YuhuiJiang99 , @shirleyzhang2 could you review this PR?

Also, while we're at it, @Rishabh-792 could you make the horizontal spacing before the code playgrounds question mark match the spacing before the Message retention question mark?

Rishabh-792 commented:
![Screenshot from 2021-12-08 12-41-51](https://user-images.githubusercontent.com/79650357/145164948-9dcc5068-9ac0-4dfb-8372-d3d92cd79e25.png)

Added the spacing. Will this work @alya ?

shirleyzhang2 commented:
@alya Thank you for bringing this to our attention and apologies for the confusion. I believe @YuhuiJiang99 was the first to claim the issue in the original thread, and we are not sure why it was claimed a second time. I guess all of us are just very excited to contribute to zulip :sweat_smile: @YuhuiJiang99 and I are first-time contributors and tried our best to follow the contribution guide, but if there is anything we could have done in the beginning to avoid this duplicate assignment please let us know!

alya commented:
I don't know why the bot let this issue get double-assigned, so hopefully it won't happen to you again. :)

Rishabh-792 commented:
@sahil839 Please review my PR. Waiting for your review and changes..

sahil839 commented:
LGTM and it's almost same as #20487 except that this PR has some more filled icons changed to empty ones.

Rishabh-792 commented:
Thanks for the review @sahil839 . 
Due to some reason, the issue got double assigned and me and @shirleyzhang2 and @YuhuiJiang99 both created a PR each addressing the issue. I am sorry for this confusion.

Rishabh-792 commented:
Hey @alya and @timabbott , requesting you to give this PR a final look and merge it if it looks good to you.

I can also encorporate changes from @shirleyzhang2 's PR (#20487 ) if needed and I think that this will also increase the quality of final commits/PR which can include all of our contributions towards the issue :)

YuhuiJiang99 commented:
@Rishabh-792 Thank you for the encorporation! Seems like the BOT got both of us unassigned without merging our PR.

Rishabh-792 commented:
@alya @sahil839 can you please check this PR a final time and please let me know about any modifications to be made to resolve the issue in a better way.

sahil839 commented:
You need to squash the commits into a single commit. Check out https://zulip.readthedocs.io/en/latest/contributing/version-control.html for guidelines.

Rishabh-792 commented:
> You need to squash the commits into a single commit and also fix the commit message. Check out https://zulip.readthedocs.io/en/latest/contributing/version-control.html for guidelines.

I squashed the commits into a single commit and also updated the commit message. 👍 
Any other suggestions ?

sahil839 commented on static/styles/settings.css:
```diff
@@ -28,6 +28,15 @@ h3 .fa-question-circle-o {
+div.settings-section-title a i {
+    opacity: 0.6;
+    margin-left: 5px;
+}
```
No need to duplicate the css here, we can just use the already existing css for `h4` or `h3` elements in this file itself.

Rishabh-792 replied:
The css here refers to the icon for "Add a new code playground" and the question icon was wrapped inside a div element and I could not use the css defined for icons under h3 and so, I created a new css specifically for the icon to change its spacing and opacity.

sahil839 replied:
So, you can just change the existing CSS as they finally have same opacity values and adding top value makes the playground icon also looks better.
```diff
--- a/static/styles/settings.css
+++ b/static/styles/settings.css
@@ -8,7 +8,8 @@ label {
 
 label,
 h4,
-h3 {
+h3,
+.settings-section-title {
     a {
         color: inherit;
     }
@@ -24,8 +25,11 @@ h3 {
     }
 }
 
-h3 .fa-question-circle-o {
-    left: 5px;
+h3,
+.settings-section-title {
+    .fa-question-circle-o {
+        left: 5px;
+    }
 }

```

Rishabh-792 replied:
Ok got it. I changed the css accordingly and will push the changes.

sahil839 commented:
LGTM apart from the point that I mentioned in other PR as well that whether we want all icons to be filled or we want to differentiate between the ones used for tooltip and those which are used as links.

Rishabh-792 commented:
> LGTM apart from the point that I mentioned in other PR as well that whether we want all icons to be filled or we want to differentiate between the ones used for tooltip and those which are used as links.

About this, Alya commented in the issue that we want the question marks to look the same preferably with empty background. 

I did this for the organization settings, and should I also change the filled icons to empty ones in other settings (user profile, stream,) too ?? @alya @sahil839

alya commented:
@Rishabh-792 I think at this point we're uncertain whether we want to do the `?` -> `i` conversion. Could you please split off your fixes for the other two points into a separate commit, so that we can get those merged? Thanks!

Rishabh-792 commented:
> @Rishabh-792 I think at this point we're uncertain whether we want to do the `?` -> `i` conversion. Could you please split off your fixes for the other two points into a separate commit, so that we can get those merged? Thanks!

I created a new PR (#21353 ) for fixing the other two tasks.

I am converting this PR into a draft until we figure out the appropriate changes for the icons.

Thanks !

alya commented:
That works!

timabbott commented:
Closing as https://github.com/zulip/zulip/pull/21353 was integrated and the rest of the issue was closed and replaced with a new plan.
